### PR TITLE
`CorrelationSimilarity` raises an error for constant data

### LINKED ISSUE #407: Raise error in CorrelationSimilarity if either column is constant
### Problem Description

_As a user, it would be help me understand the `QualityReport` better if I knew what went wrong during the computation of the correlation similarity metric._ 

### Expected behavior

- If either column in the [CorrelationSimilarity](https://github.com/sdv-dev/SDMetrics/blob/585290fc829db32645c1231d5b0385b9e90a0a4c/sdmetrics/column_pairs/statistical/correlation_similarity.py) metric is constant, we should raise an error.
- The property can catch this error and add it to the Error column.

### Additional context

Currently, if either column is constant, an warning is raised and a [dict with nan](https://github.com/sdv-dev/SDMetrics/blob/585290fc829db32645c1231d5b0385b9e90a0a4c/sdmetrics/column_pairs/statistical/correlation_similarity.py#L80) is returned. Then the property fails to read this score because we're always looking for a 'real' column [here](https://github.com/sdv-dev/SDMetrics/blob/585290fc829db32645c1231d5b0385b9e90a0a4c/sdmetrics/reports/single_table/_properties/column_pair_trends.py#L270).

### PR DESCRIPTION
**Pull request by fealho**
Resolve #407.

### CODE REVIEW

File: sdmetrics/reports/multi_table/_properties/base.py [hunk 1]
```diff
@@ -19,6 +19,23 @@ def __init__(self):
         self._properties = {}
         self.is_computed = False
 
+    def _augment_error_msg(self, table_name):
```
Comment (fealho) [inline sdmetrics/reports/multi_table/_properties/base.py:22]: Adding this to the BaseClass is not ideal, but due to the way we aggregate errors, it is the easiest way to implement it.

File: tests/integration/reports/multi_table/test_multi_table_quality_report.py [hunk 2]
```diff
@@ -164,3 +164,56 @@ def test_multi_table_quality_report():
...
+                'parent_table_name': 'table1',
+                'parent_primary_key': 'id',
+                'child_table_name': 'table2',
+                'child_foreign_key': 'id'
+            }
+        ]
+    }
+    report = QualityReport()
+
+    # Run and Assert
```
Comment (pvk-developer) [inline tests/integration/reports/multi_table/test_multi_table_quality_report.py:204]: Just # Run

**Review by amontanez24 — APPROVED** [review]
One small thing, but otherwise looks good!

File: tests/integration/reports/single_table/test_single_table_quality_report.py [hunk 3]
```diff
@@ -314,3 +314,55 @@ def test_report_with_verbose(self):
...
+
+    def test_correlation_similarity_constant_synthetic_data(self):
+        """Error out when CorrelationSimilarity is used with constant synthetic data."""
+        # Setup
+        data = pd.DataFrame({'col1': [2, 1, 1, 1], 'col2': [3, 1, 1, 1]})
+        synthetic_data = pd.DataFrame({'col1': [1, 1, 1, 1], 'col2': [1, 1, 1, 1]})
+        metadata = {'columns': {'col1': {'sdtype': 'numerical'}, 'col2': {'sdtype': 'numerical'}}}
+        report = QualityReport()
+
+        # Run and Assert
```
Comment (pvk-developer) [inline tests/integration/reports/single_table/test_single_table_quality_report.py:360]: Just # Run ?

File: tests/integration/reports/single_table/test_single_table_quality_report.py [hunk 4]
```diff
@@ -314,3 +314,55 @@ def test_report_with_verbose(self):
...
+        )
+
+    def test_correlation_similarity_one_constant_real_data_column(self):
+        """Error out when CorrelationSimilarity is used with one constant column."""
+        # Setup
+        data = pd.DataFrame({'col1': [1, 1, 1, 1], 'col2': [1.2, 1, 1, 1]})
+        metadata = {'columns': {'col1': {'sdtype': 'numerical'}, 'col2': {'sdtype': 'numerical'}}}
+        report = QualityReport()
+
+        # Run and Assert
```
Comment (pvk-developer) [inline tests/integration/reports/single_table/test_single_table_quality_report.py:342]: Just # Run ?

**Review by pvk-developer — APPROVED** [review]
LGTM!

**Pull request merged**